addons/hr_homeworking_calendar/wizard/homework_location_wizard.py

```diff
@@ -17,12 +16,32 @@ class HomeworkLocationWizard(models.TransientModel):
     weekly = fields.Boolean(default=False)
     date = fields.Date(string="Date")
     day_week_string = fields.Char(compute="_compute_day_week_string")
+    show_delete_button = fields.Boolean(compute='_show_delete_button')
 
     @api.depends('date')
     def _compute_day_week_string(self):
         for record in self:
             record.day_week_string = record.date.strftime("%A") if record.date else ''
 
+    def delete_employee_location(self):
+        self.ensure_one()
+        if not self.date or not self.employee_id:
+            return
+        employee_location = self.env['hr.employee.location'].search([
+            ('date', '=', self.date),
+            ('employee_id', '=', self.employee_id.id)
+        ])
+        if employee_location:
+            employee_location.unlink()
+
+    # Hide the delete button if record is a default recurring location
+    def _show_delete_button(self):
+        default_employee_id = self.env.context.get('default_employee_id') or self.env.user.employee_id.id
+        employee_id = self.env['hr.employee'].browse(self.employee_id.id or default_employee_id)
+        weekday = self.date.weekday()
+        default_location_for_current_date = DAYS[weekday]
+        self.show_delete_button = self.work_location_id.id != employee_id[default_location_for_current_date].id
```

Comment:
Ah, nice try, but it really seems overly complex :) Let us find a solution that is more readable on Monday.  Probably using the context this time (from the Js)